### PR TITLE
chore: update go version and packages

### PR DESCRIPTION
**Pull request by scottd018**
This will require updates to `opeator-builder` which will store a Manager and add a method `GetManager`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,81 +1,78 @@
...
-	k8s.io/client-go v0.24.2
-	sigs.k8s.io/controller-runtime v0.12.1
+	k8s.io/api v0.29.4
+	k8s.io/apiextensions-apiserver v0.29.4
+	k8s.io/apimachinery v0.29.4
+	k8s.io/client-go v0.29.4
+	sigs.k8s.io/controller-runtime v0.17.3
 )
 
+require github.com/cisco-open/k8s-objectmatcher v1.9.0
```
Comment (lander2k2) [inline go.mod]: Do we need a separate require statement just for this package?

File: go.mod [hunk 2]
```diff
@@ -1,81 +1,78 @@
...
 require (
 	emperror.dev/errors v0.8.1 // indirect
-	github.com/PuerkitoBio/purell v1.1.1 // indirect
-	github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
-	github.com/briandowns/spinner v1.18.1 // indirect
-	github.com/cespare/xxhash/v2 v2.1.2 // indirect
+	github.com/briandowns/spinner v1.23.0 // indirect
+	github.com/cespare/xxhash/v2 v2.3.0 // indirect
+	github.com/cisco-open/operator-tools v0.34.0
```
Comment (lander2k2) [inline go.mod]: Shouldn't this be included in the first require statement?  It's a direct import.

**Review by lander2k2 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (scottd018) [reply go.mod]: @lander2k2 probably not.  this is simply how `go mod tidy` wanted it formatted

Comment (scottd018) [reply go.mod]: Fixed @lander2k2

File: go.mod (hunk 2 above)
Comment (scottd018) [reply go.mod]: Yeah, go mod tidy is not kind to me today.  Fixed.

**Review by lander2k2 — APPROVED** [review]

**Pull request merged**